src/types/Typechecker/Util.hs

```diff
@@ -285,13 +285,17 @@ subtypeOf ty1 ty2
     | isStackboundType ty1 =
         liftM (isStackboundType ty2 &&) $ unbox ty1 `subtypeOf` unbox ty2
     | isArrowType ty1 && isArrowType ty2 = do
-        let argTys1 = getArgTypes ty1
-            argTys2 = getArgTypes ty2
+        let typeParams1 = getTypeParameters ty1
```

Comment:
this seems correct but I would suggest to rename `ty1` and `ty2` to `actual` and `expected` (or similar). I think it would improve the readability and ease the understanding, since I (or anyone) would need to figure out which type is what.